src/mme-app/actionHandlers/s1ReleaseActionHandlers.cpp

```diff
@@ -46,40 +46,31 @@ ActStatus ActionHandlers:: send_rel_ab_req_to_sgw(SM::ControlBlock& cb)
 	log_msg(LOG_DEBUG, "Inside send_rel_ab_req_to_sgw \n");
 
 	UEContext *ue_ctxt = dynamic_cast<UEContext*>(cb.getPermDataBlock());
-	if (ue_ctxt == NULL)
-	{
-		log_msg(LOG_DEBUG, "send_rel_ab_req_to_sgw: ue ctxt is NULL \n");
-		return ActStatus::HALT;
-	}
+	VERIFY_UE(cb, ue_ctxt, "Invalid UE\n");
 	
+    	MmeS1RelProcedureCtxt *procCtxt = dynamic_cast<MmeS1RelProcedureCtxt*>(cb.getTempDataBlock());
+    	VERIFY(procCtxt, return ActStatus::ABORT, "Procedure Context is NULL \n");
+
 	auto& sessionCtxtContainer = ue_ctxt->getSessionContextContainer();
-	if(sessionCtxtContainer.size() < 1)
-	{
-		log_msg(LOG_DEBUG, "send_rel_ab_req_to_sgw:Session context list is empty\n");
-		return ActStatus::ABORT;
-	}
+	VERIFY(sessionCtxtContainer.size() > 0,
+	        procCtxt->setMmeErrorCause(SESSION_CONTAINER_EMPTY); return ActStatus::ABORT,
+	        "Sessions Container is empty\n");
 
 	SessionContext* sessionCtxt = sessionCtxtContainer.front();
+    	VERIFY(sessionCtxt,
+            procCtxt->setMmeErrorCause(SESSION_CONTEXT_NOT_FOUND); return ActStatus::ABORT,
+            "Sessions Context is NULL\n");
 
 	BearerContext* bearerCtxt = sessionCtxt->findBearerContextByBearerId(sessionCtxt->getLinkedBearerId());
-	if (bearerCtxt == NULL)
-	{
-		log_msg(LOG_DEBUG, " send_rel_ab_req_to_sgw: bearer ctxt is NULL \n");
-		return ActStatus::HALT;
-	}
+	VERIFY(bearerCtxt,
+	        procCtxt->setMmeErrorCause(BEARER_CONTEXT_NOT_FOUND);return ActStatus::ABORT,
+	          "Bearer Context is NULL \n");
 
-    MmeS1RelProcedureCtxt *procCtxt = dynamic_cast<MmeS1RelProcedureCtxt*>(cb.getTempDataBlock());
-    if (procCtxt == NULL)
-    {
-        log_msg(LOG_DEBUG, "S1 Release Proc Ctxt is Null. Abort.\n");
+    	if(ue_ctxt->getS1apEnbUeId() != procCtxt->getS1apEnbUeId())
+    	{
+        	log_msg(LOG_DEBUG, "S1 Release req with wrong enb_s1ap_ue_id.\n");
```

Comment:
Will this cause rejection ? or silent drop at Mme?

Reply:
This gets dropped.